### ISSUE TITLE
Implement reference Dashboard as standalone SPA

### ISSUE THREAD

**Issue by sfmskywalker** [elsa-workflows/elsa-core#363]
Before we can merge Elsa 2 into master, it is paramount that we provide an alternative to the V1 Dashboard that I wish to remove in favor of a purely client side default reference application for a dashboard. This dashboard would be one of multiple other reference applications, but the primary purpose being that Elsa users who do not wish to build their own dashboard app can use it as-is, allowing them to create, list edit and delete workflows. With the availability of the GraphQL endpoint, this should be easy.

**Out of scope**
Outside the scope of this issue is the creation of an Orchard Core Elsa module, so that developers who want to create a custom dashboard can leverage existing building blocks from Orchard, such as user & role management and extend the dashboard with custom modules.

Similarly, we should create an ABP Elsa module.

**Comment by RingoKam** [comment]
oh maybe I can help with this!

I created a lightweight SPA on top of Elsa  w/ reference to the endpoint available in the razor dashboard. Do you have a preferred SPA framework in mind?

**Comment by sfmskywalker** [comment]
Great! I was thinking about using Svelte + Tailwind UI (I have a license we’re allowed to use for this project), but if you have a strong opinion to use something else then I won’t be against it.

**Comment by sfmskywalker** [comment]
I’m terms of layout/design, I think to learn from this one: https://alyssax.com/x/flowy/

**Comment by RingoKam** [comment]
I heard about Svelte but haven't use it before. it seems like an interesting idea.

Couple thoughts:

1. Does the framework plays well with GraphQL? I know React, Angular and Vue have Graphql client library. Svelte also has one https://github.com/timhall/svelte-apollo, although maybe its less popular compare to other framework implementation.
2. For other developers that want to use Elsa, is it hard for them to extend our Svelte dashboard implementation? I think Svelte is suppose to be framework agnostic since the framework exist in compile time, it doesn't have a framework runtime? 
3. I think Svelte uses rollup to build the application? Does ASP.NET Core have extensions method that plays well with Svelte? A quick google search shows this template https://github.com/NetCoreTemplates/svelte-spa/blob/master/MyApp/Startup.cs, doesn't seem like their setup support hot module reloading.

**Comment by sfmskywalker** [comment]
Those are excellent questions. I’ve spent the night awake thinking about all of this. One thing your comments made me realize is that there are several big communities out there that prefer one framework or the other. So no matter what framework we use here, there will be developers who would have preferred we went with their own framework of choice.

I’ve also been struggling with how to implement a reusable workflow designer that is “look less” so it can be configured to look whatever a developer wants it to be.

I think that ultimately there will be no one size fits all Dashboard, but rather a number of different implementations, maintained by different groups of people.

Regardless of that, we need an “official” dashboard that both non-developers as well as developers can use (and customize as they see fit).

Although there are many great framework options out there to build a modular dashboard, I realized that the one that makes me personally the most productive & happy is Blazor WASM. It has a beautiful component model like React, but arguably easier to write. Being able to leverage .NET Core is a powerful feature, where draw upon things like dependency injection and Razor components. 

I would then also propose to not be concerned with a re-hostable workflow designer right this minute, but instead implement it as a data driven templatable Razor component, with activity shape providers where developers can customize the component to be used to render a given activity on a very granular level if they wanted to. 

I think that going this route would yield an awesome product without too much effort. It’s unfortunate that Blazor doesn’t support hot reloading as of yet, but I think that’s a small price to pay for the simplicity it provides to build a modular dashboard app.

Other developers might take this project (even if just the HTML and concepts) and rewrite it in React, Vue, Angular, Alpine etc.

I don’t know what you think about all this, but you got me super-excited by letting me think about this, and now I can’t get this out of my mind 😅

If you are super excited about using a different stack, then I’d encourage you to pursuit that route. Ideally we’d  still share things like HTML markup, styling, and certain concepts where it makes sense.

But most important to me is that you have fun doing it.

Let me know your thoughts.

**Comment by mawendel** [comment]
Hey buddy, good to see Elsa move forward! My involvement in Orchard Core has stopped, at least for now, due to changing job and no longer working on that project. Anyway, i myself have been looking at standalone workflow managers and if any good designers could be found for them.

I did some research and found something promising:
https://github.com/zHaytam/Blazor.Diagrams

Looks really cool and is a port of the react-diagrams lib. Might be something that can be used here?

**Comment by GeorgeS2019** [comment]
@mawendel  and @sfmskywalker 

Yes, [Blazor Diagrams](https://github.com/zHaytam/Blazor.Diagrams/issues/41#issuecomment-727919724) look promising

**Comment by sfmskywalker** [comment]
We now have a simple dashboard application in the form of a Stencil JS component that is reusable. Although it's not yet feature complete, I will close this issue. The missing features are described https://github.com/elsa-workflows/elsa-core/issues/721.
